### ISSUE TITLE
forecast with Kalman

### ISSUE THREAD

**Issue by narencastellon** [pykalman/pykalman#114]
I have the following data:

```
ds	unique_id	y
0	2003-01-01	Bigeye Tuna	1561.149362
1	2003-01-01	Bluefin Tuna	9619.000000
2	2003-01-01	Southern Bluefin Tuna	3656.876812
3	2003-02-01	Bigeye Tuna	1507.934269
4	2003-02-01	Bluefin Tuna	8970.000000
```

How can I predict, for example, the date?
2003-02-02
2003-02-03
2003-02-04
.
.
.

this my Code

groups = tuna.groupby('unique_id')

#### Crear un filtro de Kalman
kf = KalmanFilter(transition_matrices=[1],
                  observation_matrices=[1],
                  initial_state_mean=0,
                  initial_state_covariance=1,
                  observation_covariance=1,
                  transition_covariance=0.01)

####  Aplicar el filtro de Kalman a cada grupo de datos
```
filtered_data = []
for unique_id, group in groups:
    measurements = np.asarray(group['y'])  # Convertir las mediciones a un arreglo NumPy
    kf = kf.em(measurements, n_iter = 500)
    filtered_state_means, filtered_state_covariances = kf.filter(measurements)
    smoothed_state_means, smoothed_state_covariances = kf.smooth(measurements)
    group['filtered_y'] = filtered_state_means
    group['filtered_y_smmoth'] = smoothed_state_means
    filtered_data.append(group)

#### Combinar los resultados filtrados en un solo DataFrame
filtered_data = pd.concat(filtered_data)

#### print
filtered_data
```

**Comment by fkiraly** [comment]
I don't think `pykalman` has a forecast option, but if you are using `sktime`, you can pipeline a the kalman transformers with any forecaster, that results in first applying smoothing and then the forecast on the smoothed/denoised time series.

E.g., `KalmanFilterTransformerPK(params) * ARIMA(more_params)` is a valid forecaster.

**Comment by narencastellon** [comment]
Thanks Fran, I'm going to try it. I didn't know that Sktime had a function to create a Kalman filter, is there a guide for that?

**Comment by fkiraly** [comment]
yes, it has interfaces to two packages actually, `pykalman` (now forked in `sktime` due to lapse of maintenance in the primary project, see #109), and `filterpy` (not sure about maintenance statu sthere).

The estimators are `KalmanFilterTransformerPK` and `KalmanFilterTransformerFP`, you can combine both with arbitrary forecasters.
The references would be the tutorials for forecasting, transformations, and pipelines with the two, just substitute the kalman filter transformations when you read "transformer" or "transformation".

Let me know if you need further pointers.
